File: app/src/main/java/com/eugene_poroshin/money_manager/operations/OperationsViewModel.java
package com.eugene_poroshin.money_manager.operations;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.eugene_poroshin.money_manager.repo.Repository;
import com.eugene_poroshin.money_manager.repo.database.AccountEntity;
import com.eugene_poroshin.money_manager.repo.database.Operation;
import com.eugene_poroshin.money_manager.repo.database.OperationEntity;

import java.util.List;

public class OperationsViewModel extends AndroidViewModel {

    private Repository repository;
    private LiveData<List<Operation>> allOperationsLiveData;

    public OperationsViewModel(@NonNull Application application) {
        super(application);
        repository = new Repository(application);
        allOperationsLiveData = repository.getAllOperations();
    }

    public LiveData<List<Operation>> getLiveData() {
        return allOperationsLiveData;
    }

    public void insert(OperationEntity operations) {
        repository.insert(operations);
    }

    public void update(OperationEntity operations) {
        repository.update(operations);
    }

    public void delete(OperationEntity operations) {
        repository.delete(operations);
    }
}

File: app/src/main/java/com/eugene_poroshin/money_manager/operations/OperationsViewModel.kt
package com.eugene_poroshin.money_manager.operations

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import com.eugene_poroshin.money_manager.repo.Repository
import com.eugene_poroshin.money_manager.repo.database.AppDatabase
import com.eugene_poroshin.money_manager.repo.database.Operation
import com.eugene_poroshin.money_manager.repo.database.OperationEntity
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class OperationsViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: Repository.OperationRepository
    val liveData: LiveData<List<Operation>>

    init {
        val operationDao = AppDatabase.getDatabase(application, viewModelScope).operationDao()
        repository = Repository.OperationRepository(operationDao)
        liveData = repository.allOperations
    }

    fun insert(operations: OperationEntity) = viewModelScope.launch(Dispatchers.IO) {
        repository.insert(operations)
    }

    fun update(operations: OperationEntity) = viewModelScope.launch(Dispatchers.IO) {
        repository.update(operations)
    }

    fun delete(operations: OperationEntity) = viewModelScope.launch(Dispatchers.IO) {
        repository.delete(operations)
    }
}